cunumeric/deferred.py

```diff
@@ -1740,6 +1766,28 @@ def put(self, indices: Any, values: Any, check_bounds: bool) -> None:
         if self_tmp is not self:
             self.copy(self_tmp, deep=True)
 
+    @auto_convert("mask", "values")
+    def putmask(self, mask: Any, values: Any) -> None:
+        is_scalar_value = False
+        if values.shape != self.shape and values.size == 1:
+            if values.shape == ():
+                values = values._convert_future_to_regionfield(True)
```

Comment:
is there a reason that you do this? can't you just promote this scalar store to match the shape of `self`?

Reply:
replaced this with broadcasting